### PR TITLE
new: Withdraw details keep save as log

### PR DESCRIPTION
**Pull request by alamgircsebd**
Related issue have on dokan pro https://github.com/weDevsOfficial/dokan-pro/issues/928

### CODE REVIEW

**Review by shohag121 — APPROVED** [review]

File: includes/Withdraw/Manager.php [hunk 1]
```diff
@@ -266,6 +267,22 @@ public function get_status_code( $status ) {
...
+     *
+     * @param  string $method
+     * @param  int    $user_id
+     *
+     * @return integer
+     */
+    public function get_formatted_details( $method, $user_id ) {
+        $vendor = dokan()->vendor->get( $user_id );
+
+        if ( isset( $vendor->get_payment_profiles()[$method] ) ) {
```
Comment (ManiruzzamanAkash) [inline includes/Withdraw/Manager.php:281]: We should give space before starting and ending a variable according to phpcs
`[ $method ]`

File: includes/Withdraw/Manager.php [hunk 2]
```diff
@@ -266,6 +267,22 @@ public function get_status_code( $status ) {
...
+            return [ $method => $vendor->get_payment_profiles()[$method] ];
```
Comment (ManiruzzamanAkash) [inline includes/Withdraw/Manager.php:282]: Same here - 
We should give space before starting and ending a variable according to phpcs
`[ $method ]`

**Review by ManiruzzamanAkash — APPROVED** [review]

**Comment by shanjidah** [comment]
Title: Pending withdrawal request payment gateway information changed when any vendor updates his payment information. Like vendor change bank info scb bank to Dhaka bank.

**Here is the screenshot:**
Screen Shot 2021-09-08 at 4 46 05 PM

**Comment by shanjidah** [comment]
@alamgircsebd vaiya please this above scenario.

File: includes/Upgrade/Upgrades/V_3_2_12.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+use WeDevs\Dokan\Abstracts\DokanUpgrader;
+
+class V_3_2_12 extends DokanUpgrader {
+
+    /**
+     * Update withdraw table for add new `details` column
+     *
+     * @return void
+     */
+    public static function update_dokan_withdraw_table() {
```
Comment (nurul-umbhiya) [inline includes/Upgrade/Upgrades/V_3_2_12.php:14]: fix phpcs related issues for this file

File: includes/REST/WithdrawController.php [hunk 4]
```diff
@@ -579,16 +579,20 @@ public function batch_items( $request ) {
      * @return data
      */
     public function prepare_item_for_response( $withdraw, $request ) {
-        $methods = dokan_withdraw_get_methods();
+        $methods     = dokan_withdraw_get_methods();
+        $vendor      = $this->get_user_data( $withdraw->get_user_id() );
+        $get_details = [ $withdraw->get_method() => ! empty( $withdraw->get_details() ) ? maybe_unserialize( $withdraw->get_details() ) : array() ];
```
Comment (nurul-umbhiya) [inline includes/REST/WithdrawController.php:584]: why we are using [] ? this line of code looks alien to me, can you please explain this line a bit?

File: includes/REST/WithdrawController.php [hunk 5]
```diff
@@ -579,16 +579,20 @@ public function batch_items( $request ) {
      * @return data
      */
     public function prepare_item_for_response( $withdraw, $request ) {
-        $methods = dokan_withdraw_get_methods();
+        $methods     = dokan_withdraw_get_methods();
+        $vendor      = $this->get_user_data( $withdraw->get_user_id() );
```
Comment (nurul-umbhiya) [inline includes/REST/WithdrawController.php:583]: $this->get_user_data is unsafe if actual vendor data is not found (probably user has been deleted), kindly try to rewrite it and change accordingly

**Comment by nurul-umbhiya** [comment]
My Concern is, how we've handled old data's details field! we are assigning this field to empty data, also maybe_serialized is used, but we didn't used maybe_unserialized anywhere, can you please explain it a bit?

**Review by nurul-umbhiya — APPROVED** [review]

**Pull request merged**